repl/src/main/scala/ammonite/repl/interp/Preprocessor.scala

```diff
@@ -99,30 +99,36 @@ object Preprocessor{
 
     val Expr = Processor{
       //Expressions are lifted to anon function applications so they will be JITed
-      case (name, code, tree) => Output(s"val $name = \n$code", Seq(pprint(name)))
+      case (name, code, tree) => Output(s"val $name = $code", Seq(pprint(name)))
     }
 
     val decls = Seq[(String, String, G#Tree) => Option[Preprocessor.Output]](
       ObjectDef, ClassDef, TraitDef, DefDef, TypeDef, PatVarDef, Import, Expr
     )
 
-    def apply(stmts: Seq[String], wrapperId: String): Res[Preprocessor.Output] = {
+    def apply(stmts: Seq[String],
+              wrapperId: String,
+              comment: String = ""): Res[Preprocessor.Output] = {
       val unwrapped = stmts.flatMap{x => Parsers.unwrapBlock(x) match {
-        case Some(contents) => Parsers.split(contents).get.get.value
+        case Some(contents) =>
+          Parsers.split(contents).get.get.value
+
         case None => Seq(x)
       }}
       unwrapped match{
         case Nil => Res.Skip
-        case postSplit => complete(stmts.mkString, wrapperId, postSplit.map(_.trim))
+        case postSplit =>
+          complete(stmts.mkString(""), wrapperId, postSplit, comment)
+
       }
     }
     
-    def complete(code: String, wrapperId: String, postSplit: Seq[String]) = {
+    def complete(code: String, wrapperId: String, postSplit: Seq[String], comment: String) = {
       val reParsed = postSplit.map(p => (parse(p), p))
       val errors = reParsed.collect{case (Left(e), _) => e }
       if (errors.length != 0) Res.Failure(None, errors.mkString("\n"))
       else {
-        val allDecls = for (((Right(trees), code), i) <- reParsed.zipWithIndex) yield {
+        val allDecls = for (((Right(trees @ x :: y), code), i) <- reParsed.zipWithIndex) yield {
```

Comment:
oh that makes sense. in that case could we use a `if trees.nonEmpty` check
instead?

On Friday, 27 May 2016, Abhishek Kumar <notifications@github.com> wrote:

> In repl/src/main/scala/ammonite/repl/interp/Preprocessor.scala
> <https://github.com/lihaoyi/Ammonite/pull/390#discussion_r64893350>:
>
> >        val reParsed = postSplit.map(p => (parse(p), p))
> >        val errors = reParsed.collect{case (Left(e), _) => e }
> >        if (errors.length != 0) Res.Failure(None, errors.mkString("\n"))
> >        else {
> > -        val allDecls = for (((Right(trees), code), i) <- reParsed.zipWithIndex) yield {
> > +        val allDecls = for (((Right(trees @ x :: y), code), i) <- reParsed.zipWithIndex) yield {
>
> Otherwise it can take empty trees also and was giving error accessing
> trees(0) in this line
> case trees if trees.forall(_.isInstanceOf[G#Import]) =>
> handleTree(trees(0))
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/lihaoyi/Ammonite/pull/390/files/d6b4a23f8102f8c5ac5095c2f33e4fdcb106f7b1#r64893350>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AA5A_OB6yQ3BM2wfSCZ7KIy8aCZj3gwNks5qFtmigaJpZM4Il6LB>
> .
>